### PR TITLE
feature(ui): `/allRecentJobs` page completed

### LINKED ISSUE #222: Creation of `all recent jobs` page
### Description

Currently, the `/allRecentJobs` page is not implemented as it gives us a blank response. 

#### Steps followed and expected result
Run the UI with the current backend and after login visit the All recent jobs page

PS: I would like to work on this issue

@shaheemazmalmmd @GMishx

### PR DESCRIPTION
**Pull request by soham4abc**
## Description

`/allRecentJobs` page UI is completed with a table featuring dynamic search and pagination. Also with a custom number of rows.

## Screenshot
Screenshot from 2022-06-16 08-51-35

### Changes

A new package has been installed MDBreact (https://github.com/mdbootstrap/mdb-react-ui-kit)

## How to test

install the updated dependencies and run the server to visit the `/allRecentJobs` page

Fixes #222 

@shaheemazmalmmd  @GMishx @ag4ums  
Please do have a look
(https://help.github.com/articles/closing-issues-using-keywords/)

### CODE REVIEW

**Comment by soham4abc** [comment]
The page has been updated as discussed in the meeting @ 21/06/22 

here is the video demo

https://user-images.githubusercontent.com/63705023/174948087-199b4a43-a074-485f-82b9-eb6a35282900.mp4

once approved will replicate the same in `/myRecentJobs`
@shaheemazmalmmd  @GMishx  Please do have a look.

File: src/pages/Jobs/AllJobs/index.jsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 /*
- Copyright (C) 2021 Aman Dwivedi (aman.dwivedi5@gmail.com), Shruti Agarwal (mail2shruti.ag@gmail.com)
+ Copyright (C) 2021 Aman Dwivedi (aman.dwivedi5@gmail.com), Shruti Agarwal (mail2shruti.ag@gmail.com),
+ Soham Banerjee (sohambanerjee4abc@hotmail.com)
```
Comment (GMishx) [inline src/pages/Jobs/AllJobs/index.jsx:3]: Please add the copyright year.

File: src/pages/Jobs/AllJobs/index.jsx [hunk 2]
```diff
@@ -16,20 +17,179 @@
...
+          arr[i] = res.res[i];
+        }
+        // Creating pagination
+        const arrPages = [];
+        for (let i = 0; i < res.pages; i++) {
+          arrPages.push({
+            id: i + 1,
+            value: i + 1,
+          });
+        }
```
Comment (GMishx) [inline src/pages/Jobs/AllJobs/index.jsx:81]: Are you reading the total number of pages from the response header `X-Total-Pages`?
image

File: src/pages/Jobs/AllJobs/index.jsx [hunk 3]
```diff
@@ -16,20 +17,179 @@
...
+                        <div className="font-demi">{data?.queueDate}</div>
+                      </td>
+                      <td>{data?.status}</td>
+                    </tr>
+                  ))}
+                <tr className="text-left">
+                  <td colSpan="6">
+                    Page:
+                    {pagesOptions && (
+                      <InputContainer
```
Comment (GMishx) [inline src/pages/Jobs/AllJobs/index.jsx:180]: Can pagination like this be used **iff it is not a big effort**?
image

**Review by GMishx — COMMENTED** [review]
Some queries.

File: src/pages/Jobs/AllJobs/index.jsx (hunk 2 above)
Comment (soham4abc) [reply src/pages/Jobs/AllJobs/index.jsx:81]: Yes, the number of pages is fetched from this particular header and then is read from the `res.pages`

File: src/pages/Jobs/AllJobs/index.jsx (hunk 3 above)
Comment (soham4abc) [reply src/pages/Jobs/AllJobs/index.jsx:180]: Yes we can implement this but I think this particular PR has been a very lengthy one. I plan to introduce this feature once this PR is closed. Cause once this one is closed at least the page will be introduced in the UI in a functioning state.

File: src/pages/Jobs/AllJobs/index.jsx (hunk 1 above)
Comment (soham4abc) [reply src/pages/Jobs/AllJobs/index.jsx]: Resolved

File: src/pages/Jobs/AllJobs/index.jsx (hunk 2 above)
Comment (soham4abc) [reply src/pages/Jobs/AllJobs/index.jsx:81]: From this API function:
Screenshot from 2022-06-23 12-18-34
And this services function:
Screenshot from 2022-06-23 12-18-44

File: src/pages/Jobs/AllJobs/index.jsx (hunk 3 above)
Comment (GMishx) [reply src/pages/Jobs/AllJobs/index.jsx:180]: Sure. Please create an issue for it once the PR merges.

File: src/pages/Jobs/AllJobs/index.jsx (hunk 2 above)
Comment (soham4abc) [reply src/pages/Jobs/AllJobs/index.jsx:81]: I get you.. but the local storage "pages" is also used in the browse page and is not affecting them. As we are using separate variables and states for the different pages and using useEffect each and every time a page is rendered or state is changed. It should not cause any issue. I did not find any effect of it in other pages. If you do please let me know, will look into the bug then. Thank you for your review!!

Comment (GMishx) [reply src/pages/Jobs/AllJobs/index.jsx:81]: I see it now, it is parsed in API https://github.com/fossology/FOSSologyUI/blob/183ab4ebb6e98b3856c722d2fe7e1b10e1073992/src/api/sendRequest.js#L85-L97

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**